Binary mode of NumPy fromstring removed in 2.3

**Describe the bug**
The ext tests are failing because the binary mode of NumPy's `fromstring` has been removed in 2.3. See the [release notes](https://numpy.org/doc/2.3/release/2.3.0-notes.html). Looks like we need to use `frombuffer` instead.

**To Reproduce**
```./setup test ext```

Or, because the test system insists on eating all errors and warnings:

```(cd src/tests && pytest ext_tests --import-mode=append -x)```.

**Expected behavior**
Passing tests

**Logs**
```
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.14.3, pytest-9.0.2, pluggy-1.6.0
rootdir: /home/lourens/NLeSC/projects/ReDiTSAp/activities/issue_1194_galaxia_import/amuse/src
configfile: pyproject.toml
collected 301 items                                                                                                                                                                          

ext_tests/test_boss_bodenheimer.py ..                                                                                                                                                  [  0%]
ext_tests/test_brokenimf.py ........                                                                                                                                                   [  3%]
ext_tests/test_cloud.py ..                                                                                                                                                             [  3%]
ext_tests/test_collision_handler.py .........                                                                                                                                          [  6%]
ext_tests/test_composition_methods.py ........                                                                                                                                         [  9%]
ext_tests/test_concurrent.py F

========================================================================================== FAILURES ==========================================================================================
______________________________________________________________________________ TestMPIConcurrentProcesses.test1 ______________________________________________________________________________

self = <tests.ext_tests.test_concurrent.TestMPIConcurrentProcesses testMethod=test1>

    def test1(self):
        x = ConcurrentTestingInterface(implementation_factory=Test1Implementation)
        output, error = x.do_concurrent_run()
>       self.assertEqual(error, 0, msg=output)

ext_tests/test_concurrent.py:148: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/home/lourens/.miniconda3/envs/amuse-issue-1194/lib/python3.14/site-packages/amuse/support/testing/amusetest.py:86: in failUnlessEqual
    self._raise_exceptions_if_any(failures, first, second, '{0} != {1}', msg)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <tests.ext_tests.test_concurrent.TestMPIConcurrentProcesses testMethod=test1>, failures = array([ True], dtype=bool), first = -1, second = 0, err_fmt_string = '{0} != {1}'
msg = 'The binary mode of fromstring is removed, use frombuffer instead', args = ()

    def _raise_exceptions_if_any(self, failures, first, second, err_fmt_string, msg, *args):
        if len(failures) == 1:
            if failures[0]:
>               raise self.failureException(msg or err_fmt_string.format(first, second, *args))
E               AssertionError: The binary mode of fromstring is removed, use frombuffer instead

/home/lourens/.miniconda3/envs/amuse-issue-1194/lib/python3.14/site-packages/amuse/support/testing/amusetest.py:73: AssertionError
================================================================================== short test summary info ===================================================================================
FAILED ext_tests/test_concurrent.py::TestMPIConcurrentProcesses::test1 - AssertionError: The binary mode of fromstring is removed, use frombuffer instead
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! stopping after 1 failures !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
================================================================================ 1 failed, 29 passed in 4.19s ================================================================================
```

**Environment (please complete the following information):**
numpy 2.4.2

--- a/src/amuse/ext/concurrent.py
+++ b/src/amuse/ext/concurrent.py
@@ -84,8 +84,8 @@ class MPISharedParticlesProxy(object):
         units_dump = pickle.dumps(units)
         attributes_dump = pickle.dumps(attribute_names)
         
-        units_dump = numpy.fromstring(units_dump,dtype='uint8')
-        attributes_dump = numpy.fromstring(attributes_dump,dtype='uint8')
+        units_dump = numpy.frombuffer(units_dump,dtype='uint8')
+        attributes_dump = numpy.frombuffer(attributes_dump,dtype='uint8')
         
         sendbuffer = numpy.zeros(4,  dtype='int64')
         sendbuffer[0] = self.shared_id

